### ISSUE TITLE
Cleanup SLURM/Batchsystem environment before doing builds

### ISSUE THREAD

**Issue by Flamefire** [easybuilders/easybuild-framework#4434]
Some software tests fail when run inside a SLURM job, e.g. OpenMPI which does `mpirun` that picks up the SLURM job it is running in and fails as the resulting configuration doesn't match what the job is expecting.

I have 2 workarounds in my EB wrapper script:
```
  if [[ "${SLURM_NODELIST:-}" != "" ]]; then
    ssh $SLURM_NODELIST bash -l "$0"
    exit $?
  fi
```
This basically restarts the current script via `ssh` if run from inside a SLURM job assuming only 1 node.

```
  for i in $(env | grep ^SLURM_ | cut -f1 -d=); do
    unset $i
  done
```
This removes all `SLURM_*` variables from the current environment.

As the issue is a common pitfall with EB and given how easy the 2nd variant is to implement in EB via `os.environ` I'd suggest to do this by default, possibly with a `--no-cleanup-slurm-env` option to opt-out.

**Comment by akesandgren** [comment]
Note that not all sites (like us) allow ssh between nodes in a slurm job

**Comment by Flamefire** [comment]
That was just an example. Both methods seem to work, I had the 2nd in use while our nodes where updated and didn't allow SSH yet. And only the 2nd can be reasonably done in EasyBuild.

**Comment by boegel** [comment]
Makes sense to me, and it probably makes sense to implement this in EasyBuild 5.0 (although it shouldn't actually break anything, only fix builds that break/hang because they're running in a Slurm environment which may cause trouble with MPI).

**Comment by boegel** [comment]
We briefly discussed this during the EasyBuild conf call today, and the general consensus seemed to be that this should be made opt-in rather than opt-out (which makes sense to me)

**Comment by laraPPr** [comment]
We are currently building intel/2025a and intel/2025b at our site with a lot of pain. One of the things we tried was to remove all the SLURM environment variables with:
```
unset $(env | grep ^SLURM_ | cut -f1 -d=)
```

This however caused the following error:
```
== FAILED: Installation ended unsuccessfully: Sanity check failed: sanity check command mpirun -n 8 /tmp/user/easybuild/impi/2021.15.0/intel-compilers-2025.1.1/mpi_test failed with exit code 255 (output: [mpiexec@node] Error: Unable to run bstrap_proxy on node (pid 2112056, exit code 256)
[mpiexec@node] poll_for_event (../../../../../src/pm/i_hydra/libhydra/demux/hydra_demux_poll.c:157): check exit codes error
[mpiexec@node] HYD_dmx_poll_wait_for_proxy_event (../../../../../src/pm/i_hydra/libhydra/demux/hydra_demux_poll.c:206): poll for event error
[mpiexec@node] HYD_bstrap_setup (../../../../../src/pm/i_hydra/libhydra/bstrap/src/intel/i_hydra_bstrap.c:1063): error waiting for event
[mpiexec@node] Error setting up the bootstrap proxies
[mpiexec@node] Possible reasons:
[mpiexec@node] 1. Host is unavailable. Please check that all hosts are available.
[mpiexec@node] 2. Cannot launch hydra_bstrap_proxy or it crashed on one of the hosts.
[mpiexec@node]    Make sure hydra_bstrap_proxy is available on all hosts and it has right permissions.
[mpiexec@node] 3. Firewall refused connection.
[mpiexec@node]    Check that enough ports are allowed in the firewall and specify them with the I_MPI_PORT_RANGE variable.
[mpiexec@node] 4. slurm bootstrap cannot launch processes on remote host.
[mpiexec@node]    You may try using -bootstrap option to select alternative launcher.
srun: error: Unable to confirm allocation for job 20199473: Invalid job id specified
srun: Check SLURM_JOB_ID environment variable. Expired or invalid job 20199473
```
This suggest to me that this build might require `SLURM_JOB_ID` to be set.

**Comment by akesandgren** [comment]
I would like to see an actual example of when just running things without cleaning the environment fails to work.
I've never seen such a problem here...

For impi one might need to have a hook that deals with I_MPI_PMI_LIBRARY but that's the only problem I've seen over the years.

**Comment by Thyre** [comment]
> I would like to see an actual example of when just running things without cleaning the environment fails to work. I've never seen such a problem here...
> 
> For impi one might need to have a hook that deals with I_MPI_PMI_LIBRARY but that's the only problem I've seen over the years.

Examples:
- HPCToolkit fails some tests because of SLURM environment variables being recognized, see: [HPCToolkit-2026.0.1-gompi-2026.1.eb#L65](https://github.com/easybuilders/easybuild-easyconfigs/blob/84920d4cf1a70f6a3d0fea49e7fd655ca239ddaa/easybuild/easyconfigs/h/HPCToolkit/HPCToolkit-2026.0.1-gompi-2026.1.eb#L65), https://github.com/easybuilders/easybuild-easyconfigs/pull/23830#issuecomment-3270885299, https://github.com/easybuilders/easybuild-easyconfigs/pull/23830#issuecomment-3271850192
- LINC unsets SLURM environment variables to avoid OpenMPI failures: [LINC-5.0-foss-2023b.eb#L61](https://github.com/easybuilders/easybuild-easyconfigs/blob/84920d4cf1a70f6a3d0fea49e7fd655ca239ddaa/easybuild/easyconfigs/l/LINC/LINC-5.0-foss-2023b.eb#L61)
- netCDF unsets SLURM environment variables to avoid using `srun` instead of `mpirun`: [netCDF-4.9.3-iimpi-2025a.eb#L53](https://github.com/easybuilders/easybuild-easyconfigs/blob/84920d4cf1a70f6a3d0fea49e7fd655ca239ddaa/easybuild/easyconfigs/n/netCDF/netCDF-4.9.3-iimpi-2025a.eb#L53)

And there's likely more. This was just from a quick search through the EasyConfig repository.

**Comment by akesandgren** [comment]
At least the HPCToolkit problem is due to a bad pattern that is better fixed with a patch., see https://gitlab.com/hpctoolkit/hpctoolkit/-/merge_requests/1513
